### PR TITLE
CRM-19479 : Fix for php notices

### PR DESCRIPTION
**Pull request by liedekef**
- [CRM-19479: php notices in Security.php](https://issues.civicrm.org/jira/browse/CRM-19479)

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
There are a couple of style warnings (it didn't copy very well) - we use drupal whitespace coding standards & it will trigger test fails if the whitespace does not fit with it.

Warnings
Details
File
Line
Priority
Type
Category
Security.php:400
400
High
IncorrectExact
ScopeIndent
Security.php:403
403
High
IncorrectExact
ScopeIndent

**Comment by liedekef** [comment]
Ok, I fixed the indents but now the check is failing because nothing changed???

**Comment by eileenmcnaughton** [comment]
if you click on console output you get this link

https://test.civicrm.org/job/CiviCRM-Core-PR/12082/console

 - there is an error you can see there

**Comment by liedekef** [comment]
This is embarrassing (and a prob with code only being executed once a day or so, so more difficult to test but not a valid excuse ...). If you want, I'll redo the whole thing now.

**Comment by eileenmcnaughton** [comment]
Actually, each time you add commits to the pull request (or amend & force push them) the unit tests should run again. If that isn't happened then you might need to be added to the whitelist

**Comment by liedekef** [comment]
Ok, but it seems to have passed anyway now (maybe checks aren't triggered when only whitespace changes, but still I'm sorry for the coding error in there)

**Comment by eileenmcnaughton** [comment]
@totten I've read this & it looks good - I'm pretty much ready to merge but if you could take a quick look in case you see any reason to hesitate (ie. this deals with the situation where the explode does not return 2 vars & unless we would be wildly concerned about that it seems all good)

File: CRM/Utils/Check/Component/Security.php [hunk 1]
```diff
@@ -393,8 +393,16 @@ public function guessUrl($targetDir) {
     $filePathMarker = $this->getFilePathMarker();
     $config = CRM_Core_Config::singleton();
 
-    list ($heuristicBaseUrl, $ignore) = explode($filePathMarker, $config->imageUploadURL);
-    list ($ignore, $heuristicSuffix) = explode($filePathMarker, str_replace('\\', '/', $targetDir));
+    $res_arr = explode($filePathMarker, $config->imageUploadURL);
+    $heuristicBaseUrl = $res_arr[0];
```
Comment (colemanw) [inline CRM/Utils/Check/Component/Security.php]: These two lines could be simplified as

    list ($heuristicBaseUrl) = explode($filePathMarker, $config->imageUploadURL);

File: CRM/Utils/Check/Component/Security.php [hunk 2]
```diff
@@ -393,8 +393,16 @@ public function guessUrl($targetDir) {
...
+    $res_arr = explode($filePathMarker, str_replace('\\', '/', $targetDir));
+    if (count($res_arr) == 2) {
+      $heuristicSuffix = $res_arr[1];
+    }
+    else {
+      $heuristicSuffix = "";
+    }
```
Comment (colemanw) [inline CRM/Utils/Check/Component/Security.php]: The above 7 lines could be simplified as

    $dirParts = explode($filePathMarker, str_replace('\\', '/', $targetDir));
    $heuristicSuffix = CRM_Utils_Array::value(1, $dirParts, '');

Comment (eileenmcnaughton) [inline CRM/Utils/Check/Component/Security.php]: I didn't know about the empty ',' option for list()

Comment (colemanw) [inline CRM/Utils/Check/Component/Security.php]: See example 1 in http://php.net/manual/en/function.list.php

**Comment by colemanw** [comment]
Closing in favor of #9222

**Pull request closed without merging**